src/custom/state/orders/updater.ts

```diff
@@ -180,15 +180,27 @@ export function EventUpdater(): null {
 
       const block2DateMap = await buildBlock2DateMap(library, logs)
 
-      const ordersBatchData: OrderLogPopupMixData[] = await Promise.all(
-        logs.map(async log => {
-          const { orderUid: id } = decodeTradeEvent(log)
+      // Filter out orders that should not trigger a pop-up
+      const pendingLogsAndOrders = logs.reduce<[Log, Order][]>((acc, log) => {
```

Comment:
i think in general a `filter` is better here no?

```ts
    const pendingLogsAndOrders = logs.filter<[Log, Order][]>((log) => {
        const { orderUid: id } = decodeTradeEvent(log)

        console.log(`EventUpdater::Detected Trade event for order ${id} of token in block`, log.blockNumber)

        const orderFromStore = findOrderById(id)
        if (orderFromStore && orderFromStore.status == 'pending') {
          return [log, orderFromStore]
        }
      })
```

Reply:
ah i see, cause you're transforming the return type, nvm

Reply:
I'm ok with the reduce, but I agree with the first comment of David. This could be just a filter and a map, which is easier it's easier to read

Anyways, just a nit. I'm ok leaving the reduce. I'm more concern about the complexity of the whole updater growing